### ISSUE TITLE
Pen Pressure not detected on XP-Pen tablets

### ISSUE THREAD

**Issue by relma2** [flxzt/rnote#615]
Rnote Version: 0.5.18-1
Source: Installed via AUR *and* Flatpak
System info: Endeavour OS kernel 6.2.10-arch1-1
Tablet Model: XP-Pen Deco LW
___
Pen pressure is not detected on both AUR and Flatpak downloads.
![screenshot-rnote-nopressure](https://user-images.githubusercontent.com/9358770/230802165-350fc609-4a4a-43ea-b891-375df93ef236.png)

**Comment by PhilDevProg** [comment]
Are you using X11 or Wayland? If you're using X11, is `xf86-input-wacom` installed? If you're using Wayland, are `libinput` and `libwacom` installed? And is it working on other apps like Krita?

**Comment by relma2** [comment]
xf86-input-wacom is installed, and pen pressure works fine on other applications.
Is there anything I can do to extract debug log output on my end?

**Comment by MrGumpIT** [comment]
Rnote Version: 0.5.18-1
System info: Pop!_OS 22.04 LTS 
Graphic System: Wayland
Tablet Model: XP-Pen Deco 03

You should use Wayland, it works fine for me. While before Wayland it didn’t work on Rnote or Xournal++.

**Comment by flxzt** [comment]
closing because this seems to be an issue in gtk4 stylus handling on x11, and isn't something that can be fixed by the application. I'd suggest switching to Wayland permanently, or reporting it in Gtk's bug tracker. But I wouldn't expect this to be fixed for X11 any more

**Comment by Adityashaw** [comment]
I am using Huion Tablet on `x11` without `xf86-input-wacom`. Apps like `gimp` and `xournalpp` detects pressure sensitivity properly.
I have these few questions before I try `Wayland`:
1.  Is `xf86-input-wacom` a hard requirement? If yes, then I can install and try.
2. `gtk4-demo` also doesn't show pressure sensitivity. So, shall I conclude it's `gtk4 + x11` combo issue?

**Comment by flxzt** [comment]
I am not sure if support for Huion tablets is included in `xf86-input-wacom`. There are also some "official" drivers on [Huion's website](https://www.huion.com/download/), so maybe try that as well.

But yes, apparently there is a difference in pen detection between gtk3 and gtk4.

Many users reported (see #194) that their pens are properly recognized on wayland, so I would strongly suggest to at least try it if possible.

3. did you configure that libinput is used on x11? I would refer you to the [archlinux wiki page](https://wiki.archlinux.org/title/libinput) here. On fedora this is the default (using the [xorg-x11-drv-libinput package](https://packages.fedoraproject.org/pkgs/xorg-x11-drv-libinput/xorg-x11-drv-libinput/).

**Comment by Adityashaw** [comment]
> I am not sure if support for Huion tablets is included in `xf86-input-wacom`. There are also some "official" drivers on [Huion's website](https://www.huion.com/download/), so maybe try that as well.

Yes, I'm using the official driver only. `xf86-input-wacom` doesn't support my device, hence was indirectly asking if that's the reason it doesn't work on `rnote`.

> But yes, apparently there is a difference in pen detection between gtk3 and gtk4.
> 
> Many users reported (see #194) that their pens are properly recognized on wayland, so I would strongly suggest to at least try it if possible.
> 

Sure, I'll try and update.

> 3. did you configure that libinput is used on x11? I would refer you to the [archlinux wiki page](https://wiki.archlinux.org/title/libinput) here. On fedora this is the default (using the [xorg-x11-drv-libinput package](https://packages.fedoraproject.org/pkgs/xorg-x11-drv-libinput/xorg-x11-drv-libinput/).

I did not have to configure anything related to `libinput`, it came default with `Xorg` as mentioned in the Arch wiki. I found `sudo libinput debug-events` being mentioned in one of the similar issues, so I tried it without much thought and it seemed to detect pressure sensitivity. Hence, asking if having `libinput` working enough for `rnote` or does it specifically depend on `gtk4` for pressure sensitivity (as `gtk4-demo` doesn't seem to detect it)?

**Comment by flxzt** [comment]
yes, the app depends on gtk4 for pressure sensitivity detection, so `gtk4-demo` is a good indicator for it. If that works, rnote should work as well.

**Comment by Adityashaw** [comment]
> yes, the app depends on gtk4 for pressure sensitivity detection, so `gtk4-demo` is a good indicator for it. If that works, rnote should work as well.

tried `gnome-session` on another tty with `wayland` and pressure sensitivity works.

Can we pin one of the issues or mention in `readme`?

Besides, do you have any link to the relevant `gtk4` bug so that I can go and provide some input there?
